### PR TITLE
feat: add callbacks for patient and appointment actions

### LINKED ISSUE #1855: Saving related person navigates to wrong route
## 🐛 Bug Report

After adding a new related person, the tool navigates to `/patient/:id` instead of `/patient/:id/relatedpersons`

## To Reproduce
1. Create a new patient
2. Add a new related person
3. See that tool navigates you to `/patient/:id`

## Expected behavior
Tool should navigate to `/patient/:id/relatedpersons`

## Technical notes
Instead of using the `updatePatient()` function in `patient-slice-ts`, it might be worth while creating a `addRelatedPerson()` function in `patient-slice.ts`.

Obviously, the two will share a lot of code, but we can have a helper function that has shared logic.

### PR DESCRIPTION
**Pull request by jackcmeyer**
Fixes #1855 

Also fixes similar issues for allergies and diagnoses

**Changes proposed in this pull request:**

- adds an optional onSuccess callback to each action (create, update, delete) for patients and appointments.

### CODE REVIEW

File: src/__tests__/patients/allergies/Allergies.test.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ import thunk from 'redux-thunk'
 import { Router } from 'react-router'
 import { Provider } from 'react-redux'
 import Patient from 'model/Patient'
-import { Button, Modal, List, ListItem, Alert } from '@hospitalrun/components'
+import * as components from '@hospitalrun/components'
```
Comment (fox1t) [inline src/__tests__/patients/allergies/Allergies.test.tsx]: I think here is not needed `* as components`, try with `import components from '@hospitalrun/components'`

File: src/__tests__/patients/diagnoses/Diagnoses.test.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ import Permissions from 'model/Permissions'
 import { Router } from 'react-router'
 import { Provider } from 'react-redux'
 import Diagnoses from 'patients/diagnoses/Diagnoses'
-import { Button, Modal, List, ListItem, Alert } from '@hospitalrun/components'
+import * as components from '@hospitalrun/components'
```
Comment (fox1t) [inline src/__tests__/patients/diagnoses/Diagnoses.test.tsx]: Same as above

File: src/__tests__/patients/new/NewPatient.test.tsx [hunk 3]
```diff
@@ -6,9 +6,9 @@ import { Provider } from 'react-redux'
 import { mocked } from 'ts-jest/utils'
 import { createMemoryHistory } from 'history'
 import { act } from 'react-dom/test-utils'
-import { Button } from '@hospitalrun/components'
 import configureMockStore, { MockStore } from 'redux-mock-store'
 import thunk from 'redux-thunk'
+import * as components from '@hospitalrun/components'
```
Comment (fox1t) [inline src/__tests__/patients/new/NewPatient.test.tsx]: Same as above

File: src/__tests__/patients/patient-slice.test.ts [hunk 4]
```diff
@@ -1,7 +1,6 @@
 import '../../__mocks__/matchMediaMock'
 import { AnyAction } from 'redux'
 import { mocked } from 'ts-jest/utils'
-import { createMemoryHistory } from 'history'
 import * as components from '@hospitalrun/components'
```
Comment (fox1t) [inline src/__tests__/patients/patient-slice.test.ts]: Same as above

File: src/__tests__/patients/related-persons/RelatedPersons.test.tsx [hunk 5]
```diff
@@ -4,7 +4,8 @@ import { Router } from 'react-router'
 import { createMemoryHistory } from 'history'
 import { mount } from 'enzyme'
 import RelatedPersonTab from 'patients/related-persons/RelatedPersonTab'
-import { Button, List, ListItem, Alert } from '@hospitalrun/components'
+import * as components from '@hospitalrun/components'
```
Comment (fox1t) [inline src/__tests__/patients/related-persons/RelatedPersons.test.tsx]: Same as above

File: src/__tests__/scheduling/appointments/view/ViewAppointment.test.tsx [hunk 6]
```diff
@@ -11,7 +11,7 @@ import { createMemoryHistory } from 'history'
 import AppointmentRepository from 'clients/db/AppointmentRepository'
 import { mocked } from 'ts-jest/utils'
 import { act } from 'react-dom/test-utils'
-import { Spinner, Modal } from '@hospitalrun/components'
+import * as components from '@hospitalrun/components'
```
Comment (fox1t) [inline src/__tests__/scheduling/appointments/view/ViewAppointment.test.tsx]: Same as above

File: src/__tests__/patients/allergies/Allergies.test.tsx (hunk 1 above)
Comment (jackcmeyer) [reply src/__tests__/patients/allergies/Allergies.test.tsx]: I tried this and got an error when trying to mock it.

Comment (fox1t) [reply src/__tests__/patients/allergies/Allergies.test.tsx]: OK leave it as is and we will open a new issue to investigate

**Comment by fox1t** [comment]
LGTM

**Pull request merged**